src/main/java/org/apache/mesos/state/api/StateResource.java

```diff
@@ -125,10 +131,63 @@ public Response getTaskStatus(@PathParam("taskName") String taskName) {
         } else {
             // Warning instead of Error: Subject to user input
             logger.warn(String.format("Failed to fetch requested TaskStatus for task '%s'", taskName));
-            return Response.serverError().build();
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+    }
+
+    /**
+     * Returns connection information for the given task.  The response has the following format:
+     *
+     * {
+     *     "dns": "task-name.framework-name.mesos"
+     *     "ports": "8080,2000-3000"
+     * }
+     *
+     * @param taskName Name of the task
+     * @return 200 or 404
+     */
+    @Path("/tasks/connection/{taskName}")
```

Comment:
This endpoint doesn't quite map to the VIP case or the Kafka zk case, where a single endpoint goes to all tasks. Would it make sense to also have a separate 'summary' endpoint, somewhat like the current '/connection' per-framework one, that returns a list of all tasks along with these 'global' endpoints?